cutil/cutil.go

```diff
@@ -166,18 +166,19 @@ func VerifyWorkloadVarTypes(varValue interface{}, expectedType string) error {
 			return errors.New(fmt.Sprintf("type %T, expecting %v.", varValue, expectedType))
 		}
 	case json.Number:
-		strNum := varValue.(json.Number).String()
 		if expectedType != "int" && expectedType != "float" {
 			return errors.New(fmt.Sprintf("type json.Number, expecting %v.", expectedType))
-		} else if strings.Contains(strNum, ".") && expectedType == "int" {
+		}
+		numVal, err := varValue.(json.Number).Float64()
+		if (err != nil || float64(int(numVal)) != numVal) && expectedType == "int" {
```

Comment:
Should the logic be `err != nil || (float64(int(numVal)) != numVal && expectedType == "int" )` ?